packages/simcore-sdk/src/simcore_sdk/node_ports_common/aws_s3_cli_utils.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import re
+
+from pydantic import ByteSize, parse_obj_as
+from servicelib.logging_utils import log_catch
+from servicelib.progress_bar import ProgressBarData
+
+from .r_clone_utils import BaseLogParser
+
+_logger = logging.getLogger(__name__)
+
+
+def _parse_size(log_string):
+    match = re.search(r"(\d+\.\d+|\d+)\s*(GiB|MiB|KiB|TiB|B)", log_string)
+    if match:
+        value = float(match.group(1))
+        unit = match.group(2)
+        return value, unit
+    return None, None
+
+
+class SyncAwsCliS3ProgressLogParser(BaseLogParser):
+    """
+    log processor that only yields and progress updates detected in the logs.
+
+
+    This command:
+    aws --endpoint-url http://172.17.0.1:9001 s3 sync s3://pytestbucket/d3882024-4310-472c-9b78-4e5b1d8f5dc2 . --delete --no-follow-symlinks
```

Comment:
sonarcloud will complain about this:

```suggestion
    aws --endpoint-url ENDPOINT_URL s3 sync s3://BUCKET/S3_KEY . --delete --no-follow-symlinks
```